editoast/src/views/infra/objects.rs

```diff
@@ -114,17 +120,150 @@ async fn get_objects(
+                    )
+                    .await
+                    .ok();
+                if let Some(track_section) = ts_object {
+                    results.insert(track.track.to_string(), track_section);
+                }
+            }
+            PathItemLocation::OperationalPointId { operational_point } => {
+                let op_objects = infra
+                    .get_objects(
+                        &mut db_pool.get().await?,
+                        ObjectType::OperationalPoint,
+                        &vec![operational_point.to_string()],
+                    )
+                    .await?;
+                if let Some(op) = op_objects.first() {
+                    results.insert(operational_point.to_string(), op.clone());
+                }
+            }
+            PathItemLocation::OperationalPointDescription {
+                trigram,
+                secondary_code,
+            } => {
+                let op_objects = infra
+                    .get_op_objects_by_trigram(&mut db_pool.get().await?, trigram.as_ref())
+                    .await?;
+                if let Some(op_obj) = find_op_obj_by_secondary_code(op_objects, secondary_code) {
+                    results.insert(trigram.to_string(), op_obj);
+                }
+            }
+            PathItemLocation::OperationalPointUic {
+                uic,
+                secondary_code,
+            } => {
+                let op_objects = infra
+                    .get_op_objects_by_uic(&mut db_pool.get().await?, uic as i64)
+                    .await?;
+                if let Some(op_obj) = find_op_obj_by_secondary_code(op_objects, secondary_code) {
+                    results.insert(uic.to_string(), op_obj);
+                }
+            }
+        }
+    }
+
+    Ok(Json(results))
+}
+
+fn filter_operational_point_by_ch(
+    operational_point_objects: Vec<ObjectQueryable>,
+    secondary_code: &str,
+) -> Option<ObjectQueryable> {
+    for op_obj in operational_point_objects {
+        if let Ok(op) = OperationalPoint::from_railjson(op_obj.railjson.clone()) {
+            if op.matches_ch(secondary_code) {
+                return Some(op_obj);
+            }
+        }
+    }
+    None
+}
+
+fn find_op_obj_by_secondary_code(
+    operational_point_objects: Vec<ObjectQueryable>,
+    secondary_code: Option<String>,
+) -> Option<ObjectQueryable> {
+    if operational_point_objects.is_empty() {
+        return None;
+    }
+    if let Some(secondary_code) = secondary_code {
+        filter_operational_point_by_ch(operational_point_objects, &secondary_code)
+    } else if let Some(op_obj) =
+        filter_operational_point_by_ch(operational_point_objects.clone(), CH_BV)
+    {
+        Some(op_obj)
+    } else if let Some(op_obj) =
+        filter_operational_point_by_ch(operational_point_objects.clone(), CH_00)
+    {
+        Some(op_obj)
+    } else {
+        Some(operational_point_objects.first().unwrap().clone())
```

Comment:
You could avoid the `.unwrap()` here by replacing it by something like this.
```suggestion
        Some(operational_point_objects.first()?.clone())
```
Technically, you then wouldn't need the early return at the beginning of the function testing for the emptyness of the `Vec`... but it might still be useful because it's an early-return, happening before a bunch of function calls.